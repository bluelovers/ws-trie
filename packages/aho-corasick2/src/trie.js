"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
class Trie {
    constructor() {
        this.next = {};
        this.is_word = null;
        this.value = null;
        this.data = [];
    }
    //fail?: Trie = null;
    add(word, data, original_word) {
        let chr, node;
        chr = word.charAt(0);
        node = this.next[chr];
        if (!node) {
            node = this.next[chr] = new Trie();
            if (original_word) {
                node.value = original_word.substr(0, original_word.length - word.length + 1);
            }
            else {
                node.value = chr;
            }
        }
        if (word.length > 1) {
            return node.add(word.substring(1), data, original_word || word);
        }
        else {
            node.data.push(data);
            return node.is_word = true;
        }
    }
    explore_fail_link(word) {
        let chr, i, node, _i, _ref;
        node = this;
        for (i = _i = 0, _ref = word.length; 0 <= _ref ? _i < _ref : _i > _ref; i = 0 <= _ref ? ++_i : --_i) {
            chr = word.charAt(i);
            node = node.next[chr];
            if (!node) {
                return null;
            }
        }
        return node;
    }
    each_node(callback) {
        let node, _k, _ref, _ref1;
        _ref = this.next;
        for (_k in _ref) {
            node = _ref[_k];
            callback(this, node);
        }
        _ref1 = this.next;
        for (_k in _ref1) {
            node = _ref1[_k];
            node.each_node(callback);
        }
        return this;
    }
}
exports.Trie = Trie;
exports.default = Trie;
//# sourceMappingURL=data:application/json;base64,eyJ2ZXJzaW9uIjozLCJmaWxlIjoidHJpZS5qcyIsInNvdXJjZVJvb3QiOiIiLCJzb3VyY2VzIjpbInRyaWUudHMiXSwibmFtZXMiOltdLCJtYXBwaW5ncyI6Ijs7QUFBQSxNQUFhLElBQUk7SUFBakI7UUFFQyxTQUFJLEdBRUEsRUFBRSxDQUFDO1FBQ1AsWUFBTyxHQUFZLElBQUksQ0FBQztRQUN4QixVQUFLLEdBQVcsSUFBSSxDQUFDO1FBQ3JCLFNBQUksR0FBUSxFQUFFLENBQUM7SUFpRWhCLENBQUM7SUEvREEscUJBQXFCO0lBRXJCLEdBQUcsQ0FBQyxJQUFZLEVBQUUsSUFBUSxFQUFFLGFBQXNCO1FBRWpELElBQUksR0FBVyxFQUFFLElBQVUsQ0FBQztRQUM1QixHQUFHLEdBQUcsSUFBSSxDQUFDLE1BQU0sQ0FBQyxDQUFDLENBQUMsQ0FBQztRQUNyQixJQUFJLEdBQUcsSUFBSSxDQUFDLElBQUksQ0FBQyxHQUFHLENBQUMsQ0FBQztRQUN0QixJQUFJLENBQUMsSUFBSSxFQUNUO1lBQ0MsSUFBSSxHQUFHLElBQUksQ0FBQyxJQUFJLENBQUMsR0FBRyxDQUFDLEdBQUcsSUFBSSxJQUFJLEVBQUUsQ0FBQztZQUNuQyxJQUFJLGFBQWEsRUFDakI7Z0JBQ0MsSUFBSSxDQUFDLEtBQUssR0FBRyxhQUFhLENBQUMsTUFBTSxDQUFDLENBQUMsRUFBRSxhQUFhLENBQUMsTUFBTSxHQUFHLElBQUksQ0FBQyxNQUFNLEdBQUcsQ0FBQyxDQUFDLENBQUM7YUFDN0U7aUJBRUQ7Z0JBQ0MsSUFBSSxDQUFDLEtBQUssR0FBRyxHQUFHLENBQUM7YUFDakI7U0FDRDtRQUNELElBQUksSUFBSSxDQUFDLE1BQU0sR0FBRyxDQUFDLEVBQ25CO1lBQ0MsT0FBTyxJQUFJLENBQUMsR0FBRyxDQUFDLElBQUksQ0FBQyxTQUFTLENBQUMsQ0FBQyxDQUFDLEVBQUUsSUFBSSxFQUFFLGFBQWEsSUFBSSxJQUFJLENBQUMsQ0FBQztTQUNoRTthQUVEO1lBQ0MsSUFBSSxDQUFDLElBQUksQ0FBQyxJQUFJLENBQUMsSUFBSSxDQUFDLENBQUM7WUFDckIsT0FBTyxJQUFJLENBQUMsT0FBTyxHQUFHLElBQUksQ0FBQztTQUMzQjtJQUNGLENBQUM7SUFFRCxpQkFBaUIsQ0FBQyxJQUFZO1FBRTdCLElBQUksR0FBRyxFQUFFLENBQUMsRUFBRSxJQUFhLEVBQUUsRUFBRSxFQUFFLElBQUksQ0FBQztRQUNwQyxJQUFJLEdBQUcsSUFBSSxDQUFDO1FBQ1osS0FBSyxDQUFDLEdBQUcsRUFBRSxHQUFHLENBQUMsRUFBRSxJQUFJLEdBQUcsSUFBSSxDQUFDLE1BQU0sRUFBRSxDQUFDLElBQUksSUFBSSxDQUFDLENBQUMsQ0FBQyxFQUFFLEdBQUcsSUFBSSxDQUFDLENBQUMsQ0FBQyxFQUFFLEdBQUcsSUFBSSxFQUFFLENBQUMsR0FBRyxDQUFDLElBQUksSUFBSSxDQUFDLENBQUMsQ0FBQyxFQUFFLEVBQUUsQ0FBQyxDQUFDLENBQUMsRUFBRSxFQUFFLEVBQ25HO1lBQ0MsR0FBRyxHQUFHLElBQUksQ0FBQyxNQUFNLENBQUMsQ0FBQyxDQUFDLENBQUM7WUFDckIsSUFBSSxHQUFHLElBQUksQ0FBQyxJQUFJLENBQUMsR0FBRyxDQUFDLENBQUM7WUFDdEIsSUFBSSxDQUFDLElBQUksRUFDVDtnQkFDQyxPQUFPLElBQUksQ0FBQzthQUNaO1NBQ0Q7UUFDRCxPQUFPLElBQUksQ0FBQztJQUNiLENBQUM7SUFFRCxTQUFTLENBQUMsUUFBNkM7UUFFdEQsSUFBSSxJQUFJLEVBQUUsRUFBRSxFQUFFLElBQUksRUFBRSxLQUFLLENBQUM7UUFDMUIsSUFBSSxHQUFHLElBQUksQ0FBQyxJQUFJLENBQUM7UUFDakIsS0FBSyxFQUFFLElBQUksSUFBSSxFQUNmO1lBQ0MsSUFBSSxHQUFHLElBQUksQ0FBQyxFQUFFLENBQUMsQ0FBQztZQUNoQixRQUFRLENBQUMsSUFBSSxFQUFFLElBQUksQ0FBQyxDQUFDO1NBQ3JCO1FBQ0QsS0FBSyxHQUFHLElBQUksQ0FBQyxJQUFJLENBQUM7UUFDbEIsS0FBSyxFQUFFLElBQUksS0FBSyxFQUNoQjtZQUNDLElBQUksR0FBRyxLQUFLLENBQUMsRUFBRSxDQUFDLENBQUM7WUFDakIsSUFBSSxDQUFDLFNBQVMsQ0FBQyxRQUFRLENBQUMsQ0FBQztTQUN6QjtRQUNELE9BQU8sSUFBSSxDQUFDO0lBQ2IsQ0FBQztDQUNEO0FBeEVELG9CQXdFQztBQUVELGtCQUFlLElBQUksQ0FBQSIsInNvdXJjZXNDb250ZW50IjpbImV4cG9ydCBjbGFzcyBUcmllPFQgPSBhbnk+XG57XG5cdG5leHQ6IHtcblx0XHRbazogc3RyaW5nXTogVHJpZTxUPlxuXHR9ID0ge307XG5cdGlzX3dvcmQ6IGJvb2xlYW4gPSBudWxsO1xuXHR2YWx1ZTogc3RyaW5nID0gbnVsbDtcblx0ZGF0YTogVFtdID0gW107XG5cblx0Ly9mYWlsPzogVHJpZSA9IG51bGw7XG5cblx0YWRkKHdvcmQ6IHN0cmluZywgZGF0YT86IFQsIG9yaWdpbmFsX3dvcmQ/OiBzdHJpbmcpOiBib29sZWFuXG5cdHtcblx0XHRsZXQgY2hyOiBzdHJpbmcsIG5vZGU6IFRyaWU7XG5cdFx0Y2hyID0gd29yZC5jaGFyQXQoMCk7XG5cdFx0bm9kZSA9IHRoaXMubmV4dFtjaHJdO1xuXHRcdGlmICghbm9kZSlcblx0XHR7XG5cdFx0XHRub2RlID0gdGhpcy5uZXh0W2Nocl0gPSBuZXcgVHJpZSgpO1xuXHRcdFx0aWYgKG9yaWdpbmFsX3dvcmQpXG5cdFx0XHR7XG5cdFx0XHRcdG5vZGUudmFsdWUgPSBvcmlnaW5hbF93b3JkLnN1YnN0cigwLCBvcmlnaW5hbF93b3JkLmxlbmd0aCAtIHdvcmQubGVuZ3RoICsgMSk7XG5cdFx0XHR9XG5cdFx0XHRlbHNlXG5cdFx0XHR7XG5cdFx0XHRcdG5vZGUudmFsdWUgPSBjaHI7XG5cdFx0XHR9XG5cdFx0fVxuXHRcdGlmICh3b3JkLmxlbmd0aCA+IDEpXG5cdFx0e1xuXHRcdFx0cmV0dXJuIG5vZGUuYWRkKHdvcmQuc3Vic3RyaW5nKDEpLCBkYXRhLCBvcmlnaW5hbF93b3JkIHx8IHdvcmQpO1xuXHRcdH1cblx0XHRlbHNlXG5cdFx0e1xuXHRcdFx0bm9kZS5kYXRhLnB1c2goZGF0YSk7XG5cdFx0XHRyZXR1cm4gbm9kZS5pc193b3JkID0gdHJ1ZTtcblx0XHR9XG5cdH1cblxuXHRleHBsb3JlX2ZhaWxfbGluayh3b3JkOiBzdHJpbmcpXG5cdHtcblx0XHRsZXQgY2hyLCBpLCBub2RlOiBUcmllPFQ+LCBfaSwgX3JlZjtcblx0XHRub2RlID0gdGhpcztcblx0XHRmb3IgKGkgPSBfaSA9IDAsIF9yZWYgPSB3b3JkLmxlbmd0aDsgMCA8PSBfcmVmID8gX2kgPCBfcmVmIDogX2kgPiBfcmVmOyBpID0gMCA8PSBfcmVmID8gKytfaSA6IC0tX2kpXG5cdFx0e1xuXHRcdFx0Y2hyID0gd29yZC5jaGFyQXQoaSk7XG5cdFx0XHRub2RlID0gbm9kZS5uZXh0W2Nocl07XG5cdFx0XHRpZiAoIW5vZGUpXG5cdFx0XHR7XG5cdFx0XHRcdHJldHVybiBudWxsO1xuXHRcdFx0fVxuXHRcdH1cblx0XHRyZXR1cm4gbm9kZTtcblx0fVxuXG5cdGVhY2hfbm9kZShjYWxsYmFjazogKHRyaWU6IHRoaXMsIG5vZGU6IFRyaWU8VD4pID0+IHZvaWQpXG5cdHtcblx0XHRsZXQgbm9kZSwgX2ssIF9yZWYsIF9yZWYxO1xuXHRcdF9yZWYgPSB0aGlzLm5leHQ7XG5cdFx0Zm9yIChfayBpbiBfcmVmKVxuXHRcdHtcblx0XHRcdG5vZGUgPSBfcmVmW19rXTtcblx0XHRcdGNhbGxiYWNrKHRoaXMsIG5vZGUpO1xuXHRcdH1cblx0XHRfcmVmMSA9IHRoaXMubmV4dDtcblx0XHRmb3IgKF9rIGluIF9yZWYxKVxuXHRcdHtcblx0XHRcdG5vZGUgPSBfcmVmMVtfa107XG5cdFx0XHRub2RlLmVhY2hfbm9kZShjYWxsYmFjayk7XG5cdFx0fVxuXHRcdHJldHVybiB0aGlzO1xuXHR9XG59XG5cbmV4cG9ydCBkZWZhdWx0IFRyaWVcbiJdfQ==