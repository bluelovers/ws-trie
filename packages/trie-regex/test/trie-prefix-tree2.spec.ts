import trie from 'trie-prefix-tree2';
import trieToRegExp from '../index';

const arr = ['trea', 'tr2a', 'trie', '1', 'foobar', 'foobaz', 'foozap', 'fooza', '$'];

test(`array`, () =>
{
	let t1 = trie(arr).tree();

	let re = trieToRegExp(t1);

	arr.forEach(c =>
	{
		expect(c).toMatch(re);
	})

	expect(re).toMatchSnapshot();
	expect(t1).toMatchSnapshot();

});

test(`emoji`, () =>
{
	let arr = ["😀","😃","😄","😁","😆","😅","🤣","😂","🙂","🙃","😉","😊","😇","🥰","😍","🤩","😘","😗","☺️","☺","😚","😙","😋","😛","😜","🤪","😝","🤑","🤗","🤭","🤫","🤔","🤐","🤨","😐","😑","😶","😏","😒","🙄","😬","🤥","😌","😔","😪","🤤","😴","😷","🤒","🤕","🤢","🤮","🤧","🥵","🥶","🥴","😵","🤯","🤠","🥳","😎","🤓","🧐","😕","😟","🙁","☹️","☹","😮","😯","😲","😳","🥺","😦","😧","😨","😰","😥","😢","😭","😱","😖","😣","😞","😓","😩","😫","🥱","😤","😡","😠","🤬","😈","👿","💀","☠️","☠","💩","🤡","👹","👺","👻","👽","👾","🤖","😺","😸","😹","😻","😼","😽","🙀","😿","😾","🙈","🙉","🙊","💋","💌","💘","💝","💖","💗","💓","💞","💕","💟","❣️","❣",];

	let t1 = trie(arr).tree();

	let re = trieToRegExp(t1);

	arr.forEach(c =>
	{
		expect(c).toMatch(re);
	})

	expect(re).toMatchSnapshot();
	expect(t1).toMatchSnapshot();

});
