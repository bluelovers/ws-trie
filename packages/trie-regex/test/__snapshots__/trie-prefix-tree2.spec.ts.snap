// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`array 1`] = `/\\(\\?:foo\\(\\?:ba\\[rz\\]\\|zap\\?\\)\\|tr\\(\\?:2a\\|ea\\|ie\\)\\|\\\\\\$\\|1\\)/u`;

exports[`array 2`] = `
{
  "$": {
    "$$": {
      "$": null,
      Symbol(default): "$",
    },
  },
  "1": {
    "$$": {
      "1": null,
      Symbol(default): "1",
    },
  },
  "f": {
    "o": {
      "o": {
        "b": {
          "a": {
            "r": {
              "$$": {
                "foobar": null,
                Symbol(default): "foobar",
              },
            },
            "z": {
              "$$": {
                "foobaz": null,
                Symbol(default): "foobaz",
              },
            },
          },
        },
        "z": {
          "a": {
            "$$": {
              "fooza": null,
              Symbol(default): "fooza",
            },
            "p": {
              "$$": {
                "foozap": null,
                Symbol(default): "foozap",
              },
            },
          },
        },
      },
    },
  },
  "t": {
    "r": {
      "2": {
        "a": {
          "$$": {
            "tr2a": null,
            Symbol(default): "tr2a",
          },
        },
      },
      "e": {
        "a": {
          "$$": {
            "trea": null,
            Symbol(default): "trea",
          },
        },
      },
      "i": {
        "e": {
          "$$": {
            "trie": null,
            Symbol(default): "trie",
          },
        },
      },
    },
  },
}
`;

exports[`emoji 1`] = `/\\(\\?:\\\\u263A\\\\uFE0F\\|\\\\u2620\\\\uFE0F\\|\\\\u2639\\\\uFE0F\\|\\\\u2763\\\\uFE0F\\|\\\\u\\{1F4A9\\}\\|\\\\u\\{1F9D0\\}\\|\\\\u\\{1F47A\\}\\|\\\\u\\{1F47B\\}\\|\\\\u\\{1F47D\\}\\|\\\\u\\{1F47E\\}\\|\\\\u\\{1F47F\\}\\|\\\\u\\{1F48B\\}\\|\\\\u\\{1F48C\\}\\|\\\\u\\{1F49D\\}\\|\\\\u\\{1F49E\\}\\|\\\\u\\{1F49F\\}\\|\\\\u\\{1F60A\\}\\|\\\\u\\{1F60B\\}\\|\\\\u\\{1F60C\\}\\|\\\\u\\{1F60D\\}\\|\\\\u\\{1F60E\\}\\|\\\\u\\{1F60F\\}\\|\\\\u\\{1F61A\\}\\|\\\\u\\{1F61B\\}\\|\\\\u\\{1F61C\\}\\|\\\\u\\{1F61D\\}\\|\\\\u\\{1F61E\\}\\|\\\\u\\{1F61F\\}\\|\\\\u\\{1F62A\\}\\|\\\\u\\{1F62B\\}\\|\\\\u\\{1F62C\\}\\|\\\\u\\{1F62D\\}\\|\\\\u\\{1F62E\\}\\|\\\\u\\{1F62F\\}\\|\\\\u\\{1F63A\\}\\|\\\\u\\{1F63B\\}\\|\\\\u\\{1F63C\\}\\|\\\\u\\{1F63D\\}\\|\\\\u\\{1F63E\\}\\|\\\\u\\{1F63F\\}\\|\\\\u\\{1F64A\\}\\|\\\\u\\{1F92A\\}\\|\\\\u\\{1F92B\\}\\|\\\\u\\{1F92C\\}\\|\\\\u\\{1F92D\\}\\|\\\\u\\{1F92E\\}\\|\\\\u\\{1F92F\\}\\|\\\\u\\{1F97A\\}\\|\\\\u\\{1F479\\}\\|\\\\u\\{1F480\\}\\|\\\\u\\{1F493\\}\\|\\\\u\\{1F495\\}\\|\\\\u\\{1F496\\}\\|\\\\u\\{1F497\\}\\|\\\\u\\{1F498\\}\\|\\\\u\\{1F600\\}\\|\\\\u\\{1F601\\}\\|\\\\u\\{1F602\\}\\|\\\\u\\{1F603\\}\\|\\\\u\\{1F604\\}\\|\\\\u\\{1F605\\}\\|\\\\u\\{1F606\\}\\|\\\\u\\{1F607\\}\\|\\\\u\\{1F608\\}\\|\\\\u\\{1F609\\}\\|\\\\u\\{1F610\\}\\|\\\\u\\{1F611\\}\\|\\\\u\\{1F612\\}\\|\\\\u\\{1F613\\}\\|\\\\u\\{1F614\\}\\|\\\\u\\{1F615\\}\\|\\\\u\\{1F616\\}\\|\\\\u\\{1F617\\}\\|\\\\u\\{1F618\\}\\|\\\\u\\{1F619\\}\\|\\\\u\\{1F620\\}\\|\\\\u\\{1F621\\}\\|\\\\u\\{1F622\\}\\|\\\\u\\{1F623\\}\\|\\\\u\\{1F624\\}\\|\\\\u\\{1F625\\}\\|\\\\u\\{1F626\\}\\|\\\\u\\{1F627\\}\\|\\\\u\\{1F628\\}\\|\\\\u\\{1F629\\}\\|\\\\u\\{1F630\\}\\|\\\\u\\{1F631\\}\\|\\\\u\\{1F632\\}\\|\\\\u\\{1F633\\}\\|\\\\u\\{1F634\\}\\|\\\\u\\{1F635\\}\\|\\\\u\\{1F636\\}\\|\\\\u\\{1F637\\}\\|\\\\u\\{1F638\\}\\|\\\\u\\{1F639\\}\\|\\\\u\\{1F640\\}\\|\\\\u\\{1F641\\}\\|\\\\u\\{1F642\\}\\|\\\\u\\{1F643\\}\\|\\\\u\\{1F644\\}\\|\\\\u\\{1F648\\}\\|\\\\u\\{1F649\\}\\|\\\\u\\{1F910\\}\\|\\\\u\\{1F911\\}\\|\\\\u\\{1F912\\}\\|\\\\u\\{1F913\\}\\|\\\\u\\{1F914\\}\\|\\\\u\\{1F915\\}\\|\\\\u\\{1F916\\}\\|\\\\u\\{1F917\\}\\|\\\\u\\{1F920\\}\\|\\\\u\\{1F921\\}\\|\\\\u\\{1F922\\}\\|\\\\u\\{1F923\\}\\|\\\\u\\{1F924\\}\\|\\\\u\\{1F925\\}\\|\\\\u\\{1F927\\}\\|\\\\u\\{1F928\\}\\|\\\\u\\{1F929\\}\\|\\\\u\\{1F970\\}\\|\\\\u\\{1F971\\}\\|\\\\u\\{1F973\\}\\|\\\\u\\{1F974\\}\\|\\\\u\\{1F975\\}\\|\\\\u\\{1F976\\}\\|\\\\u263A\\|\\\\u2620\\|\\\\u2639\\|\\\\u2763\\)/u`;

exports[`emoji 2`] = `
{
  "☠": {
    "$$": {
      "☠": null,
      Symbol(default): "☠",
    },
  },
  "☠️": {
    "$$": {
      "☠️": null,
      Symbol(default): "☠️",
    },
  },
  "☹": {
    "$$": {
      "☹": null,
      Symbol(default): "☹",
    },
  },
  "☹️": {
    "$$": {
      "☹️": null,
      Symbol(default): "☹️",
    },
  },
  "☺": {
    "$$": {
      "☺": null,
      Symbol(default): "☺",
    },
  },
  "☺️": {
    "$$": {
      "☺️": null,
      Symbol(default): "☺️",
    },
  },
  "❣": {
    "$$": {
      "❣": null,
      Symbol(default): "❣",
    },
  },
  "❣️": {
    "$$": {
      "❣️": null,
      Symbol(default): "❣️",
    },
  },
  "👹": {
    "$$": {
      "👹": null,
      Symbol(default): "👹",
    },
  },
  "👺": {
    "$$": {
      "👺": null,
      Symbol(default): "👺",
    },
  },
  "👻": {
    "$$": {
      "👻": null,
      Symbol(default): "👻",
    },
  },
  "👽": {
    "$$": {
      "👽": null,
      Symbol(default): "👽",
    },
  },
  "👾": {
    "$$": {
      "👾": null,
      Symbol(default): "👾",
    },
  },
  "👿": {
    "$$": {
      "👿": null,
      Symbol(default): "👿",
    },
  },
  "💀": {
    "$$": {
      "💀": null,
      Symbol(default): "💀",
    },
  },
  "💋": {
    "$$": {
      "💋": null,
      Symbol(default): "💋",
    },
  },
  "💌": {
    "$$": {
      "💌": null,
      Symbol(default): "💌",
    },
  },
  "💓": {
    "$$": {
      "💓": null,
      Symbol(default): "💓",
    },
  },
  "💕": {
    "$$": {
      "💕": null,
      Symbol(default): "💕",
    },
  },
  "💖": {
    "$$": {
      "💖": null,
      Symbol(default): "💖",
    },
  },
  "💗": {
    "$$": {
      "💗": null,
      Symbol(default): "💗",
    },
  },
  "💘": {
    "$$": {
      "💘": null,
      Symbol(default): "💘",
    },
  },
  "💝": {
    "$$": {
      "💝": null,
      Symbol(default): "💝",
    },
  },
  "💞": {
    "$$": {
      "💞": null,
      Symbol(default): "💞",
    },
  },
  "💟": {
    "$$": {
      "💟": null,
      Symbol(default): "💟",
    },
  },
  "💩": {
    "$$": {
      "💩": null,
      Symbol(default): "💩",
    },
  },
  "😀": {
    "$$": {
      "😀": null,
      Symbol(default): "😀",
    },
  },
  "😁": {
    "$$": {
      "😁": null,
      Symbol(default): "😁",
    },
  },
  "😂": {
    "$$": {
      "😂": null,
      Symbol(default): "😂",
    },
  },
  "😃": {
    "$$": {
      "😃": null,
      Symbol(default): "😃",
    },
  },
  "😄": {
    "$$": {
      "😄": null,
      Symbol(default): "😄",
    },
  },
  "😅": {
    "$$": {
      "😅": null,
      Symbol(default): "😅",
    },
  },
  "😆": {
    "$$": {
      "😆": null,
      Symbol(default): "😆",
    },
  },
  "😇": {
    "$$": {
      "😇": null,
      Symbol(default): "😇",
    },
  },
  "😈": {
    "$$": {
      "😈": null,
      Symbol(default): "😈",
    },
  },
  "😉": {
    "$$": {
      "😉": null,
      Symbol(default): "😉",
    },
  },
  "😊": {
    "$$": {
      "😊": null,
      Symbol(default): "😊",
    },
  },
  "😋": {
    "$$": {
      "😋": null,
      Symbol(default): "😋",
    },
  },
  "😌": {
    "$$": {
      "😌": null,
      Symbol(default): "😌",
    },
  },
  "😍": {
    "$$": {
      "😍": null,
      Symbol(default): "😍",
    },
  },
  "😎": {
    "$$": {
      "😎": null,
      Symbol(default): "😎",
    },
  },
  "😏": {
    "$$": {
      "😏": null,
      Symbol(default): "😏",
    },
  },
  "😐": {
    "$$": {
      "😐": null,
      Symbol(default): "😐",
    },
  },
  "😑": {
    "$$": {
      "😑": null,
      Symbol(default): "😑",
    },
  },
  "😒": {
    "$$": {
      "😒": null,
      Symbol(default): "😒",
    },
  },
  "😓": {
    "$$": {
      "😓": null,
      Symbol(default): "😓",
    },
  },
  "😔": {
    "$$": {
      "😔": null,
      Symbol(default): "😔",
    },
  },
  "😕": {
    "$$": {
      "😕": null,
      Symbol(default): "😕",
    },
  },
  "😖": {
    "$$": {
      "😖": null,
      Symbol(default): "😖",
    },
  },
  "😗": {
    "$$": {
      "😗": null,
      Symbol(default): "😗",
    },
  },
  "😘": {
    "$$": {
      "😘": null,
      Symbol(default): "😘",
    },
  },
  "😙": {
    "$$": {
      "😙": null,
      Symbol(default): "😙",
    },
  },
  "😚": {
    "$$": {
      "😚": null,
      Symbol(default): "😚",
    },
  },
  "😛": {
    "$$": {
      "😛": null,
      Symbol(default): "😛",
    },
  },
  "😜": {
    "$$": {
      "😜": null,
      Symbol(default): "😜",
    },
  },
  "😝": {
    "$$": {
      "😝": null,
      Symbol(default): "😝",
    },
  },
  "😞": {
    "$$": {
      "😞": null,
      Symbol(default): "😞",
    },
  },
  "😟": {
    "$$": {
      "😟": null,
      Symbol(default): "😟",
    },
  },
  "😠": {
    "$$": {
      "😠": null,
      Symbol(default): "😠",
    },
  },
  "😡": {
    "$$": {
      "😡": null,
      Symbol(default): "😡",
    },
  },
  "😢": {
    "$$": {
      "😢": null,
      Symbol(default): "😢",
    },
  },
  "😣": {
    "$$": {
      "😣": null,
      Symbol(default): "😣",
    },
  },
  "😤": {
    "$$": {
      "😤": null,
      Symbol(default): "😤",
    },
  },
  "😥": {
    "$$": {
      "😥": null,
      Symbol(default): "😥",
    },
  },
  "😦": {
    "$$": {
      "😦": null,
      Symbol(default): "😦",
    },
  },
  "😧": {
    "$$": {
      "😧": null,
      Symbol(default): "😧",
    },
  },
  "😨": {
    "$$": {
      "😨": null,
      Symbol(default): "😨",
    },
  },
  "😩": {
    "$$": {
      "😩": null,
      Symbol(default): "😩",
    },
  },
  "😪": {
    "$$": {
      "😪": null,
      Symbol(default): "😪",
    },
  },
  "😫": {
    "$$": {
      "😫": null,
      Symbol(default): "😫",
    },
  },
  "😬": {
    "$$": {
      "😬": null,
      Symbol(default): "😬",
    },
  },
  "😭": {
    "$$": {
      "😭": null,
      Symbol(default): "😭",
    },
  },
  "😮": {
    "$$": {
      "😮": null,
      Symbol(default): "😮",
    },
  },
  "😯": {
    "$$": {
      "😯": null,
      Symbol(default): "😯",
    },
  },
  "😰": {
    "$$": {
      "😰": null,
      Symbol(default): "😰",
    },
  },
  "😱": {
    "$$": {
      "😱": null,
      Symbol(default): "😱",
    },
  },
  "😲": {
    "$$": {
      "😲": null,
      Symbol(default): "😲",
    },
  },
  "😳": {
    "$$": {
      "😳": null,
      Symbol(default): "😳",
    },
  },
  "😴": {
    "$$": {
      "😴": null,
      Symbol(default): "😴",
    },
  },
  "😵": {
    "$$": {
      "😵": null,
      Symbol(default): "😵",
    },
  },
  "😶": {
    "$$": {
      "😶": null,
      Symbol(default): "😶",
    },
  },
  "😷": {
    "$$": {
      "😷": null,
      Symbol(default): "😷",
    },
  },
  "😸": {
    "$$": {
      "😸": null,
      Symbol(default): "😸",
    },
  },
  "😹": {
    "$$": {
      "😹": null,
      Symbol(default): "😹",
    },
  },
  "😺": {
    "$$": {
      "😺": null,
      Symbol(default): "😺",
    },
  },
  "😻": {
    "$$": {
      "😻": null,
      Symbol(default): "😻",
    },
  },
  "😼": {
    "$$": {
      "😼": null,
      Symbol(default): "😼",
    },
  },
  "😽": {
    "$$": {
      "😽": null,
      Symbol(default): "😽",
    },
  },
  "😾": {
    "$$": {
      "😾": null,
      Symbol(default): "😾",
    },
  },
  "😿": {
    "$$": {
      "😿": null,
      Symbol(default): "😿",
    },
  },
  "🙀": {
    "$$": {
      "🙀": null,
      Symbol(default): "🙀",
    },
  },
  "🙁": {
    "$$": {
      "🙁": null,
      Symbol(default): "🙁",
    },
  },
  "🙂": {
    "$$": {
      "🙂": null,
      Symbol(default): "🙂",
    },
  },
  "🙃": {
    "$$": {
      "🙃": null,
      Symbol(default): "🙃",
    },
  },
  "🙄": {
    "$$": {
      "🙄": null,
      Symbol(default): "🙄",
    },
  },
  "🙈": {
    "$$": {
      "🙈": null,
      Symbol(default): "🙈",
    },
  },
  "🙉": {
    "$$": {
      "🙉": null,
      Symbol(default): "🙉",
    },
  },
  "🙊": {
    "$$": {
      "🙊": null,
      Symbol(default): "🙊",
    },
  },
  "🤐": {
    "$$": {
      "🤐": null,
      Symbol(default): "🤐",
    },
  },
  "🤑": {
    "$$": {
      "🤑": null,
      Symbol(default): "🤑",
    },
  },
  "🤒": {
    "$$": {
      "🤒": null,
      Symbol(default): "🤒",
    },
  },
  "🤓": {
    "$$": {
      "🤓": null,
      Symbol(default): "🤓",
    },
  },
  "🤔": {
    "$$": {
      "🤔": null,
      Symbol(default): "🤔",
    },
  },
  "🤕": {
    "$$": {
      "🤕": null,
      Symbol(default): "🤕",
    },
  },
  "🤖": {
    "$$": {
      "🤖": null,
      Symbol(default): "🤖",
    },
  },
  "🤗": {
    "$$": {
      "🤗": null,
      Symbol(default): "🤗",
    },
  },
  "🤠": {
    "$$": {
      "🤠": null,
      Symbol(default): "🤠",
    },
  },
  "🤡": {
    "$$": {
      "🤡": null,
      Symbol(default): "🤡",
    },
  },
  "🤢": {
    "$$": {
      "🤢": null,
      Symbol(default): "🤢",
    },
  },
  "🤣": {
    "$$": {
      "🤣": null,
      Symbol(default): "🤣",
    },
  },
  "🤤": {
    "$$": {
      "🤤": null,
      Symbol(default): "🤤",
    },
  },
  "🤥": {
    "$$": {
      "🤥": null,
      Symbol(default): "🤥",
    },
  },
  "🤧": {
    "$$": {
      "🤧": null,
      Symbol(default): "🤧",
    },
  },
  "🤨": {
    "$$": {
      "🤨": null,
      Symbol(default): "🤨",
    },
  },
  "🤩": {
    "$$": {
      "🤩": null,
      Symbol(default): "🤩",
    },
  },
  "🤪": {
    "$$": {
      "🤪": null,
      Symbol(default): "🤪",
    },
  },
  "🤫": {
    "$$": {
      "🤫": null,
      Symbol(default): "🤫",
    },
  },
  "🤬": {
    "$$": {
      "🤬": null,
      Symbol(default): "🤬",
    },
  },
  "🤭": {
    "$$": {
      "🤭": null,
      Symbol(default): "🤭",
    },
  },
  "🤮": {
    "$$": {
      "🤮": null,
      Symbol(default): "🤮",
    },
  },
  "🤯": {
    "$$": {
      "🤯": null,
      Symbol(default): "🤯",
    },
  },
  "🥰": {
    "$$": {
      "🥰": null,
      Symbol(default): "🥰",
    },
  },
  "🥱": {
    "$$": {
      "🥱": null,
      Symbol(default): "🥱",
    },
  },
  "🥳": {
    "$$": {
      "🥳": null,
      Symbol(default): "🥳",
    },
  },
  "🥴": {
    "$$": {
      "🥴": null,
      Symbol(default): "🥴",
    },
  },
  "🥵": {
    "$$": {
      "🥵": null,
      Symbol(default): "🥵",
    },
  },
  "🥶": {
    "$$": {
      "🥶": null,
      Symbol(default): "🥶",
    },
  },
  "🥺": {
    "$$": {
      "🥺": null,
      Symbol(default): "🥺",
    },
  },
  "🧐": {
    "$$": {
      "🧐": null,
      Symbol(default): "🧐",
    },
  },
}
`;
